hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java

```diff
@@ -667,6 +668,11 @@ public void handleJoinRequest(JoinRequest joinRequest, Connection connection) {
             if (now > timeToStartJoin) {
                 startJoin();
             }
+            if (node.isMaster()) {
+                long startTime = clusterClock.getClusterStartTime();
+                nodeEngine.getOperationService().send(new ClusterStartTimeUpdateOperation(startTime),
```

Comment:
Why do you need another operation to set cluster start time, instead of putting this information into finalize join operation?